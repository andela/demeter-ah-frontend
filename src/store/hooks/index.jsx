import { useEffect } from 'react';
import { getUser } from '../actions/signup';

export const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return [];
};
