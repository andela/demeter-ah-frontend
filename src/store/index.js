import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import combineReducers from './reducers';
import dummyAction from './actions/Signup';

const middlewares = applyMiddleware(thunk);

const store = createStore(
  combineReducers,
  composeWithDevTools(middlewares),
);

store.dispatch(dummyAction);

export default store;
