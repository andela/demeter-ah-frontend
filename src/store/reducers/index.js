import { combineReducers } from 'redux';
import auth from './Auth';
import articleListing from './articleListing';
import resetPassword from './resetPassword';
import changePassword from './changePassword';
import articles from './articles';
import editProfile from './editProfile';
import viewProfile from './viewProfile';
import membership from './Membership';
import viewArticle from './viewArticle';
import relatedArticles from './relatedArticles';
import bookmarks from './bookmarks';
import bookmarkArticle from './bookmarkArticle';
import comments from './Comments';
import search from './search';
import notifications from './notifications';
import rate from './rateArticle';

const reducers = combineReducers({
  auth,
  resetPassword,
  changePassword,
  articles,
  editProfile,
  viewProfile,
  membership,
  articleListing,
  viewArticle,
  relatedArticles,
  bookmarks,
  bookmarkArticle,
  comments,
  search,
  notifications,
  rate
});

export default reducers;
