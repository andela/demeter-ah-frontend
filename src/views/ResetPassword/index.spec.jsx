import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../store';
import ReduxResetPassword, { ResetPasswordComp } from './index';

describe('RESET PASSWORD COMPONENT', () => {
  it('Should render without errors', () => {
    const component = shallow(
      <Router>
        <ReduxResetPassword store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  it('Should render along with children componenet', () => {
    const component = mount(
      <Router>
        <ReduxResetPassword store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
    expect(component.find('button')).toHaveLength(1);
    expect(component.find('input')).toHaveLength(1);
  });

  it('should call onChange function', () => {
    const event = {
      preventDefault() { },
      target: { value: 'frank@gmail.com', name: 'email' }
    };
    const mockOnResetFn = jest.fn();
    const component = mount(
      <ResetPasswordComp authResponse={{ error: 'bad data' }} cleanUpReset={() => { }} isSubmitting={false} sendResetLink={() => { }} onSignUp={mockOnResetFn} />
    );
    const inputTag = component.find('input').at(0);
    inputTag.simulate('change', event);
  });

  it('should call handleSubmit function', () => {
    const mockOnResetFn = jest.fn();
    const fakeEvent = { preventDefault: mockOnResetFn };
    const component = shallow(
      <ResetPasswordComp authResponse={{ message: 'welcome' }} cleanUpReset={() => { }} isSubmitting={false} sendResetLink={() => { }} onSignUp={mockOnResetFn} />
    );
    component.find('form').simulate('submit', fakeEvent);
    expect(mockOnResetFn.mock.calls.length).toBe(1);
  });

  it('should change button value from reset to loading on submit', () => {
    const component = mount(
      <ResetPasswordComp authResponse={{ message: 'welcome' }} cleanUpReset={() => { }} isSubmitting sendResetLink={() => { }} />
    );
    expect(component.find('button').text()).toEqual('Loading');
  });
});
