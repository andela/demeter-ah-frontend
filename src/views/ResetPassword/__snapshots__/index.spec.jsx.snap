// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RESET PASSWORD COMPONENT Should render along with children componenet 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(ResetPassword)
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <ResetPassword
        authResponse={Object {}}
        cleanUpReset={[Function]}
        isSubmitting={false}
        sendResetLink={[Function]}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <main
          className="reset"
        >
          <form
            autoComplete="off"
            className="lg:w-5/12 md:w-3/5 sm:w-4/5 w-full min-w-min sm:rounded-2xl rounded-none"
            onSubmit={[Function]}
          >
            <h3
              className="text-center text-white text-3xl font-semibold"
            >
              Reset Password
            </h3>
            <InputForm
              classes="my-8 md:w-3/5 w-4/5 mx-auto"
              inputType="email"
              labelname="Email"
              name="email"
              onChange={[Function]}
              pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
              placeholder="johndoe@gmail.com"
            >
              <div
                className="my-8 md:w-3/5 w-4/5 mx-auto"
              >
                <Label
                  labelname="Email"
                  name="email"
                >
                  <label
                    htmlFor="email"
                  >
                    Email
                  </label>
                </Label>
                <Input
                  inputType="email"
                  name="email"
                  onChange={[Function]}
                  pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                  placeholder="johndoe@gmail.com"
                >
                  <input
                    className="forminput rounded py-2 bg-purple-150 undefined"
                    name="email"
                    onChange={[Function]}
                    pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                    placeholder="johndoe@gmail.com"
                    required={true}
                    type="email"
                  />
                </Input>
              </div>
            </InputForm>
            <Button
              className="submitBtn"
              isSubmit={false}
              type="submit"
            >
              <button
                className="submitBtn"
                disabled={false}
                type="submit"
              >
                Reset
              </button>
            </Button>
          </form>
        </main>
      </ResetPassword>
    </Connect(ResetPassword)>
  </Router>
</BrowserRouter>
`;

exports[`RESET PASSWORD COMPONENT Should render status view after reset success 1`] = `
<ResetPassword
  authResponse={
    Object {
      "message": "welcome",
    }
  }
  cleanUpReset={[Function]}
  isSubmitting={false}
  sendResetLink={[Function]}
>
  <main
    className="reset"
  >
    <Status
      height="h-86"
      status="welcome"
      width="w-2/4"
    >
      <div
        className="w-2/4 h-86 container"
      >
        <h3
          className="status"
        >
          welcome
        </h3>
        <div
          className="actions"
        />
      </div>
    </Status>
  </main>
</ResetPassword>
`;

exports[`RESET PASSWORD COMPONENT Should render without errors 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(ResetPassword)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</Router>
`;
